### PR TITLE
Allow consul and nomad syslog

### PR DESCRIPTION
**Pull request by guangie88**
Also minor reorder parts of draining section so that the instructions is always to check for members in ASG, before the actual drain command.

### CODE REVIEW

File: modules/core/packer/roles/nomad/files/templates/syslog.hcl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (lawliet89) [inline modules/core/packer/roles/nomad/files/templates/syslog.hcl]: This file is HCL. Don't need to be in JSON format. Don't need the quotes in keys and the braces.

```hcl
enable_syslog = {{ enable_syslog | to_json }}
```

File: modules/core/packer/roles/nomad/tasks/main.yml [hunk 2]
```diff
@@ -11,11 +11,18 @@
     version: "{{ nomad_module_version }}"
 - name: Install Nomad
   shell: "/tmp/terraform-aws-nomad/modules/install-nomad/install-nomad --version {{ nomad_version }} --user {{ nomad_user }}"
-- name: "Copy Nomad Configuration"
+- name: "Copy Nomad configuration"
   copy:
     src: "{{ role_path }}/files/config/"
     dest: "/opt/nomad/config"
   become: yes
+- name: "Copy Nomad syslog sub-configuration"
```
Comment (lawliet89) [inline modules/core/packer/roles/nomad/tasks/main.yml]: Maybe just call this `template` configuration?

File: modules/core/packer/roles/nomad/files/templates/syslog.hcl (hunk 1 above)
Comment (guangie88) [reply modules/core/packer/roles/nomad/files/templates/syslog.hcl]: Good catch, fixed

File: modules/core/packer/roles/nomad/tasks/main.yml (hunk 2 above)
Comment (guangie88) [reply modules/core/packer/roles/nomad/tasks/main.yml]: Okay used `with_items` to go with the template configuration idea, did the same for `Consul` side.

**Pull request merged**